src/index/FTSAlgorithms.cpp

```diff
@@ -306,48 +302,72 @@ void FTSAlgorithms::aggScoresAndTakeTopKContexts(
       << " elements to a table with distinct entities "
       << "and at most " << k << " contexts per entity.\n";
 
+  size_t numOfTerms = dynResult->numColumns() - 3;
```

Comment:
Shouldn't you read the number of terms from the `wep` and then CHECK() that the `dynResult` has the correct number of columns?
Actually we should just build the `IdTable` from scratch in this function, but that is a story for another day.

Reply:
And just call it `numTerms`